### PR TITLE
feat: add temperature interface

### PR DESCRIPTION
**Pull request by chocol4te**
Virtually copy-pasted from @jonas-schievink's PR, https://github.com/nrf-rs/nrf51-hal/pull/17.

### CODE REVIEW

File: nrf52-hal-common/src/temp.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /// If a measurement was already started, it will be canceled.
+    pub fn measure(&mut self) -> I30F2 {
+        self.stop_measurement();
+        self.start_measurement();
+
+        while self.0.events_datardy.read().bits() == 0 {}
+
+        self.0.events_datardy.reset(); // clear event
+        let raw = self.0.temp.read().bits();
+        I30F2::from_bits(raw as i32)
```
Comment (hannobraun) [inline nrf52-hal-common/src/temp.rs]: Why not call `read` in a loop instead? This seems to be the same code, basically.

Comment (jonas-schievink) [reply nrf52-hal-common/src/temp.rs]: Yes, this can just use `nb::block!`

Comment (chocol4te) [reply nrf52-hal-common/src/temp.rs]: Done :)

**Comment by chocol4te** [comment]
@hannobraun I have tested this on 2 nRF52810's, I do not have any other chips but the datasheets indicate they behave identically on all variants.

**Comment by Yatekii** [comment]
I can test the nRF52840 in the evening/this weekend. The question is how you verify that the readings are correct ;)

**Comment by chocol4te** [comment]
Well I put one in the freezer 🤷‍♂️

**Comment by Yatekii** [comment]
Ah, I wrongfully assumed we want to veryfiy the accuracy, which of course isn't necessary to verify correct driver functionality. Will test ;)

**Comment by jonas-schievink** [comment]
I've checked the datasheets and the nrf52 version is equivalent in data format and interface to the nrf51 sensor, except that the nrf52 chips also have additional registers for linearity compensation (which we don't use, and don't have to use).

**Comment by chocol4te** [comment]
We can open an issue for adding that additional functionality, if there's anyone who needs it

File: nrf52-hal-common/src/temp.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            self.0.events_datardy.reset();
+        }
+    }
+
+    /// Tries to read a started measurement (non-blocking).
+    ///
+    /// Before calling this, `start_measurement` must be called.
+    ///
+    /// Returns the measured temperature in °C.
+    pub fn read(&mut self) -> nb::Result<I30F2, ()> {
```
Comment (hannobraun) [inline nrf52-hal-common/src/temp.rs]: It seems like this function can't fail (except for `WouldBlock`) and `()` is not the right type to represent this. I don't think using the never type (`!`) in this position is stable yet, but `void::Void` would work. I believe `void` is already a dependency of this crate.

```suggestion
    pub fn read(&mut self) -> nb::Result<I30F2, Void> {
```
Note: This suggestion only covers the change required in this line. `Void` would also need to be imported above..

Comment (chocol4te) [reply nrf52-hal-common/src/temp.rs]: Done :)

**Comment by chocol4te** [comment]
I'm not sure how to mark your suggestion as added,  I just thought it would be easier to manually commit :/

**Pull request merged**